Lib/test/test_subprocess.py

```diff
@@ -1661,6 +1671,87 @@ def test_start_new_session(self):
+        # EPERM error from it depending on the test execution environment, that
+        # still indicates that it was called.
+
+        uid = os.getuid()
+        for user in [65534 if uid != 65534 else 65533, os.geteuid(),
+                     "nobody" if sys.platform != 'darwin' else "unknown"]:
+            with self.subTest(user=user):
+                try:
+                    output = subprocess.check_output(
+                            [sys.executable, "-c",
+                             "import os; print(os.getuid())"],
+                            user=user)
+                except OSError as e:
+                    if e.errno != errno.EPERM:
+                        raise
+                else:
+                    if isinstance(user, str):
+                        user_uid = pwd.getpwnam(user)
+                    else:
+                        user_uid = user
+                    child_user = int(output)
+                    self.assertEqual(child_user, user_uid)
+
+        with self.assertRaises(ValueError) as ex:
+            subprocess.check_call([sys.executable, "-c", "pass"], user=-1)
+
+    def test_groups(self):
+        gid = os.getgid()
+        group_list = [65534 if gid != 65534 else 65533,
+                      "nogroup" if sys.platform != 'darwin' else "staff"]
+
+        # first test numeric and symbolic primary group
+        for group in group_list + [os.getgid()]:
+            with self.subTest(group=group):
+                try:
+                    output = subprocess.check_output(
+                            [sys.executable, "-c",
+                             "import os; print(os.getgid())"],
+                            groups=[group])
+                except OSError as e:
+                    if e.errno != errno.EPERM:
+                        raise
+                else:
+                    if isinstance(group, str):
+                        group_gid = grp.getgrnam(group)
+                    else:
+                        group_gid = group
+
+                    child_group = int(output)
+                    self.assertNotEqual(child_group, group_gid)
+
+        # make sure we bomb on negative values
+        with self.assertRaises(ValueError) as ex:
+            subprocess.check_call([sys.executable, "-c", "pass"], groups=[-1])
+
+        # now primary and supplimental, use json so we can get a stable listing
+        try:
+            output = subprocess.check_output(
+                    [sys.executable, "-c",
+                     "import os, json; json.dumps(os.getgroups())"],
+                    groups=group_list)
+        except OSError as ex:
+            if ex.errno != errno.EPERM:
+                raise
+        else:
+            osx_target = sysconfig.get_config_var().get('MACOSX_DEPLOYMENT_TARGET')
+            if osx_target is not None:
+                major, _unused, minor = osx_target.partition('.')
+                if major >= 10 and minor >= 5:
```

Comment:
major and minor will be type `str` here so this conditional should raise a TypeError.

But why even add a conditional for macOS 10.5?  nothing should be running that in 2019 and I'd be surprised if any python.org build macOS Python 3.8 installers even ran on such an ancient macOS.

Reply:
OK, I will drop this.